mentoring/mentoring.py

```diff
@@ -156,14 +162,34 @@ class MentoringBlock(XBlock, StepParentMixin, StudioEditableXBlockMixin, StudioC
     has_score = True
     has_children = True
 
+    block_settings_key = 'mentoring'
+    theme_key = 'theme'
+
     def _(self, text):
         """ translate text """
         return self.runtime.service(self, "i18n").ugettext(text)
 
     @property
     def is_assessment(self):
+        """ Checks if mentoring XBlock is in assessment mode """
         return self.mode == 'assessment'
 
+    def get_theme(self):
+        """
+        Gets theme settings from settings service. Falls back to default (LMS) theme
+        if settings service is not available, xblock theme settings are not set or does
+        contain mentoring theme settings.
+        """
+        settings_service = self.runtime.service(self, "settings")
+        if not settings_service:
+            return _default_theme_config
+
+        xblock_settings = settings_service.get_settings_bucket(self, {})
+        if not xblock_settings or not hasattr(xblock_settings, 'get'):
+            return _default_theme_config
+
+        return xblock_settings.get(self.theme_key, _default_theme_config)
```

Comment:
When is xblock_settings truthy but doesn't have a `get()` attribute?

Anyhow, how about this? I think it's more DRY:
```python
settings_service = self.runtime.service(self, "settings")
if settings_service:
    xblock_settings = settings_service.get_settings_bucket(self)
    if xblock_settings and self.theme_key in xblock_settings:
        return xblock_settings[self.theme_key]
return _default_theme_config
```